src/main/java/pro/trevor/tankgame/state/board/unit/tank/GenericTank.java

```diff
@@ -0,0 +1,155 @@
+package pro.trevor.tankgame.state.board.unit.tank;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import pro.trevor.tankgame.rule.type.IPlayerElement;
+import pro.trevor.tankgame.rule.type.ITickElement;
+import pro.trevor.tankgame.state.board.IMovable;
+import pro.trevor.tankgame.state.board.IPositioned;
+import pro.trevor.tankgame.state.board.Position;
+import pro.trevor.tankgame.state.board.unit.tank.status.IAttributeStatus;
+import pro.trevor.tankgame.state.board.unit.tank.status.IStatus;
+
+import java.util.*;
+
+public class GenericTank<E extends Enum<E> & IAttribute> implements IMovable, IPositioned, ITickElement, IPlayerElement {
+
+    protected final String player;
+    protected Position position;
+    protected final Map<E, Object> attributes;
+    protected final Set<IStatus> statuses;
+
+    public GenericTank(String player, Position position, Map<E, Object> defaults) {
+        this.player = player;
+        this.position = position;
+        this.attributes = new HashMap<>();
+        this.statuses = new HashSet<>();
+        for (E attribute : defaults.keySet()) {
+            attributes.put(attribute, defaults.get(attribute));
+        }
+    }
+
+    protected <T> T get(E attribute, Class<T> type) {
+        if (attribute.getType().isAssignableFrom(type)) {
+            try {
+                return type.cast(attributes.get(attribute));
+            } catch (ClassCastException ignored) {
+                throw new Error(String.format("Attribute %s was not stored as a(n) %s", attribute.name(), type.getSimpleName()));
+            }
+        } else {
+            throw new Error(String.format("Unable to read attribute %s as a(n) %s", attribute.name(), type.getSimpleName()));
+        }
+    }
+
+    void set(E attribute, Object object) {
+        if (attribute.getType().isAssignableFrom(object.getClass())) {
+            attributes.put(attribute, object);
+        } else {
+            throw new Error(String.format("Attribute %s cannot store a(n) %s", attribute.name(), object.getClass().getSimpleName()));
+        }
+    }
+
+    int getInteger(E attribute) {
```

Comment:
get does not apply status effects but getInteger does.  It's probably a good idea to update the name to indicate which methods apply status effects so that it's clear in code calling these functions.